Code/Tools/AssetProcessor/native/AssetManager/SourceFileRelocator.cpp

```diff
@@ -1025,14 +1025,14 @@ Please note that only those seed files will get updated that are active for your
         AZStd::binary_semaphore waitSignal;
         int errorCount = 0;
 
-        AzToolsFramework::SourceControlResponseCallbackBulk callback = AZStd::bind(&HandleSourceControlResult,
-            AZStd::ref (relocationContainer),
-            AZStd::ref(waitSignal),
-            AZStd::ref(errorCount),
-            static_cast<AzToolsFramework::SourceControlFlags>(SCF_OpenByUser), // If a file is moved from A -> B and then again from B -> A, the result is just an "edit", so we're just going to assume success if the file is checked out, regardless of state
-            true,
-            AZStd::placeholders::_1,
-            AZStd::placeholders::_2);
+        AzToolsFramework::SourceControlResponseCallbackBulk callback = [&](bool success, AZStd::vector<SourceControlFileInfo> info)
+            {
```

Comment:
Have you tested this with Perforce integration? This change makes me nervous because I don't think anyone's regularly testing Perforce integration with the engine in git now, and this change directly touches Perforce integration code.

Reply:
Is perforce integration a feature we want to maintain?

Reply:
I can't see how converting bind to a lambda could impact perforce integration at all, especially since SourceControlResponseCallbackBulk is `AZStd::function` which should consume any callable ?

Reply:
> Is perforce integration a feature we want to maintain?
Even if we're using git now, Perforce is standard in the games industry, and last I heard it's still supported for our tools.

> I can't see how converting bind to a lambda could impact perforce integration at all, especially since SourceControlResponseCallbackBulk is AZStd::function which should consume any callable ?
I don't expect the change to impact it, but we've been burned by that before with seemingly safe changes breaking things.

Reply:
Is there some kind of test I could run to verify this, without buying Perforce?

Reply:
Not that I'm aware of, I think all Perforce testing is currently manual right now, and is pretty time consuming. I'll click approve for now, but I'm still nervous about this changing this logic due to the infrequency of testing in this area.

Reply:
The logic shouldn't have changed at all. Simple replacement of older library construct with a language one.

Reply:
Calling @lumberyard-employee-dm who can probably sanity check this for correctness (and will be happy to see an `AZStd::bind` usage removed 🙂)

Reply:
To be truly safe, what should occur is to not use `&` for capturing the local variable parameters by reference, but to capture each argument as part of the lambda capture.

```
[&relocationContainer, &waitSignal, &errorCount, ](bool success, AZStd::vector<SourceControlFileInfo> info)
```

Reply:
To be honest, I've followed the most common way lambda is used around the codebase, and in this case, capturing by`&` should be semantically the same?

Reply:
It should be, but this is to prevent "over" capture of local variables

Reply:
I agree that 'blanket' `&` in a lambda's capture list can cause problems, but, in this case, the lambda's body is quite small, not like for example the lambda in `Remapper<IdType, AllowDuplicates>::RemapIds` :smile: 

So while I can easily rewrite it as suggested, there are ~600 instances of `[&]` capture lambdas around the codebase that will likely need similar treatment?

Reply:
We should keep code that is working as is.
So we shouldn't proactively go out of the way to have "lambdas only capture what they use".
But over capture can cause unnecessary memory allocations when stored in containers like a std::function, which contains a small buffer that can store captured function objects within a small buffer internally.

Reply:
Oh, I fully agree and understand that keeping things lean by design is a good goal. I'll update that lambda capture tomorrow, to contain only used variables, and would like to suggest updating the https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md to include this guidance. (@hultonha, @lawsonamzn  ? )

Reply:
The `&` capture isn't blocking this change from going in.
I have no issues with it as it stands and because the change has made through AR and has enough approvals I am going to merge it to the development.

Reply:
> 
> 
> Oh, I fully agree and understand that keeping things lean by design is a good goal. I'll update that lambda capture tomorrow, to contain only used variables, and would like to suggest updating the https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md to include this guidance. (@hultonha, @lawsonamzn ? )

That's a great idea @nemerle, might be worth seeing if there's somewhere for that recommendation to go if we have a section on lambdas (I think the other thing too would be to bring this up in the sig-core Discord to discuss - you can now start a thread about it there which is cool! 😄)